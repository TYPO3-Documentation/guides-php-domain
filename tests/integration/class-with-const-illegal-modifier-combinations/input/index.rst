========================
PHP class with constants
========================

..  php:class:: ChuckleFactory\FunClass

    Where Fun Knows No Bounds!

    ..  php:const:: JOY_CONSTANT
        :public:
        :private:

        The constant that encapsulates pure joy and laughter. Whenever you need
        a pick-me-up, just access JOY_CONSTANT and let the chuckles begin.

    ..  php:const:: GIGGLE_FACTOR
        :public:
        :protected:

        A dynamic constant that measures the giggle intensity in the ChuckleFactory.
        It's known to spontaneously increase during code reviews and coffee breaks.

    ..  php:const:: WHIMSICAL_WONDER
        :protected:
        :private:

        Unleash the power of whimsy! This constant is your key to unlocking the
        whimsical wonders hidden within the FunClass. Expect surprises and delight!
