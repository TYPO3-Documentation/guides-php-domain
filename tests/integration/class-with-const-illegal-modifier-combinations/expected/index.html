<!-- content start -->
    <div class="section" id="php-class-with-constants">
    <h1>PHP class with constants</h1>

            <dl class="php class">
    <dt class="sig sig-object php"
        id="chucklefactory-funclass">
        <em class="property"><span class="pre">class</span> </em>
            <span class="sig-prename descclassname"><span class="pre">\ChuckleFactory\</span></span>
<span class="sig-name descname"><span class="pre">FunClass</span></span>
</dt>
    <dd>
        <p>Where Fun Knows No Bounds!</p><dl class="php const">
    <dt class="sig sig-object php" id="chucklefactory-funclass-joy-constant">
        <span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>
 <em class="property"><span class="pre">const</span></em>
        <span class="sig-name descname"><span class="pre">JOY_CONSTANT</span></span>

    </dt>
    <dd><p>The constant that encapsulates pure joy and laughter. Whenever you need
a pick-me-up, just access JOY_CONSTANT and let the chuckles begin.</p></dd>
</dl>
<dl class="php const">
    <dt class="sig sig-object php" id="chucklefactory-funclass-giggle-factor">
        <span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="sig-name modifier"><span class="pre">protected</span></span>
 <em class="property"><span class="pre">const</span></em>
        <span class="sig-name descname"><span class="pre">GIGGLE_FACTOR</span></span>

    </dt>
    <dd><p>A dynamic constant that measures the giggle intensity in the ChuckleFactory.
It&#039;s known to spontaneously increase during code reviews and coffee breaks.</p></dd>
</dl>
<dl class="php const">
    <dt class="sig sig-object php" id="chucklefactory-funclass-whimsical-wonder">
        <span class="sig-name modifier"><span class="pre">protected</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>
 <em class="property"><span class="pre">const</span></em>
        <span class="sig-name descname"><span class="pre">WHIMSICAL_WONDER</span></span>

    </dt>
    <dd><p>Unleash the power of whimsy! This constant is your key to unlocking the
whimsical wonders hidden within the FunClass. Expect surprises and delight!</p></dd>
</dl>

    </dd>
</dl>

    </div>

<!-- content end -->
